imagetocsv/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"os"
+	"strconv"
+	"strings"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+)
+
+func readImage(filename string) image.Image {
+	fileReader, err := os.Open(filename)
+	if err != nil {
+		panic(err)
+	}
+	img, _, err := image.Decode(fileReader)
+	if err != nil {
+		panic(err)
+	}
+	return img
+}
+
+// ignore alpha for now
+func colorToBrightness(c color.Color) int {
+	r, g, b, _ := c.RGBA()
+	return (int)((r + g + b) / 3)
```

Comment:
For grey scale images you usually weight the colour values according to how sensitive human eyes are to the different colours. If we use that weighting the formula is:
`(0.2125 * color.r) + (0.7154 * color.g) + (0.0721 * color.b)`